### DISCUSSION TITLE
Missing files when building the project<rc_autosar_rh850>

### DISCUSSION THREAD

**Discussion by decoski** [toppers/users-forum#102] [Q&A] [answered]
Hello:
When I use trunk-473(https://dev.toppers.jp/trac_user/contrib/browser/rc_autosar_rh850/#trunk/tools) to build the project atk2-sc1, I prompt that the following files are missing:

(E)	F0520005	E:\TOPPER-Files\trunk-473\trunk\atk2-sc1\include\Os.h(83):F0520005:Could not open source file "Os_Cfg.h"	Os.h	83	atk2-sc1.mtpj
(E)	E0511134	E0511134:Input file "cfg\Os_Lcfg_asm.asm" is not found.			atk2-sc1.mtpj
(E)	E0511134	E0511134:Input file "cfg\Os_Lcfg.c" is not found.			atk2-sc1.mtpj

My environment is CS+CC V8.00; atk2-sc1-v1.4.0 ;
a-rtegen V1.2.2; a-comstackV1.2.1; a-can V1.2.1.

How to get the missing documents，Waiting for your answer, thank you！

**Comment by decoski** [comment]
@mitsut Sorry to bother you, can you answer this question“#102”?

**Comment by mitsut** [comment]
Looking at the error message, it seems that `cfg.exe` is not working.
Please check two points of interest.

1) Check the CS+ plugin.
From the CS+ menu > Tools > Manage Plugins and check that the IronPython Consource Plug-in is ticked.
Image attached.
![Notification_Center](https://user-images.githubusercontent.com/2193540/217974056-d0fa9d79-974b-4d29-919e-26c234b383a2.png)

2) The second is the version of CS+. We have not checked that it works with CS+ version 7 or later. This may be due to the fact that your version is 8.

**Reply by decoski** [reply]
Thank you for your reply. I generated the file<Os_ Cfg. h>when compiling<os app>, presumably cfg.exe has already run. However, the prompt kernel.lib cannot be found. It seems that it is still a generation problem. At the same time, CS+prompts that R7F701035xAFP cannot be found_ common.xml.
I don't have this file, so I use R7F701034xAFP_ Common.xml (CS+FOR CC v8.00 exists) and rename it as 1035 ， then the computer is restarted, and kernel.lib is generated. It is speculated that the default device of ATK2-sc1 is R7F701035. Is that right???

Similarly, I am generating C:  TOPPER  run  swc application  **2ecu_ co_ b\ecu_ co**
This sub-project still exists
(E)	F0563300	F0563300:Cannot open file : "kernel\DefaultBuild\kernel.lib"			atk2-sc1.mtpj

Please help me to solve such mistakes, thank you!

**Comment by mitsut** [comment, marked as answer]
Thank you for the information, kernel.lib is not generated?
Please confirm one more point of concern.
The CS+ project consists of two subprojects cfg and kernel subprojects and atk2-sc1 project.
Is the atk2-sc1 project active?
If the cfg subproject is active, then select the atk2-sc1 project and set it as active from the menu.
![スクリーンショット 2023-02-12 093222](https://user-images.githubusercontent.com/2193540/218287089-ad107bd7-eb28-4ea1-a768-fb147e11e1cf.png)

If it is already active, the menu should be inactive.

**Reply by decoski** [reply]
Thank you for your reply. I did a test with the trunk**\swc-application\2ecu_cod_b** project and found that the prompt was missing kernel.lib or Os_ Cfg. h requires two subprojects kernel or  cfg project  build first,then you can build the project-atk2-sc1 successfully. The above is for reference.

**Comment by mitsut** [comment]
> I don't have this file, so I use R7F701034xAFP_ Common.xml (CS+FOR CC v8.00 exists) and rename it as 1035 ， then the computer is restarted, and kernel.lib is generated. It is speculated that the default device of ATK2-sc1 is R7F701035. Is that right???

Yes, You are right way.
For CS+, the corresponding chip number changes with each version.
We had checked up to Ver6, but Version8 requires similar work.

**Reply by decoski** [reply]
If it is convenient, please also answer # 104
